### PR TITLE
fix(nemesis.py): move decommission nemesis to a function

### PR DESCRIPTION
**Pull request by fgelcer**
because we have to change target node in case we
run decommission seed node, we are probably messing with the `is_seed` property, and making this nemesis fail/break when called.
Fixes: #5433

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fgelcer** [comment]
@yarongilor , please create a staging job using this branch, run only this nemesis in the 1tb job (wait some 5 iterations of it), and if it solved the issue, we can merge this one.

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -1211,6 +1211,9 @@ def disrupt_nodetool_decommission(self, add_node=True, disruption_name=None):
                 self.unset_current_running_nemesis(new_node)
         return new_node
 
+    def disrupt_nodetool_decommission(self, add_node=True, disruption_name=None):
+        return self._nodetool_decommission()
```
Comment (vponomaryov) [inline sdcm/nemesis.py:1215]: `return self._nodetool_decommission(add_node=add_node, disruption_name=disruption_name)`

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: sdcm/nemesis.py (hunk 1 above)
Comment (fgelcer) [reply sdcm/nemesis.py]: you are completely right, missed it in here.
fixed

**Review by vponomaryov — DISMISSED** [review]
LGTM

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -1211,6 +1211,9 @@ def disrupt_nodetool_decommission(self, add_node=True, disruption_name=None):
                 self.unset_current_running_nemesis(new_node)
         return new_node
 
+    def disrupt_nodetool_decommission(self, add_node=True, disruption_name=None):
+        return self._nodetool_decommission(add_node=add_node, disruption_name=disruption_name)
```
Comment (fruch) [inline sdcm/nemesis.py:1215]: `disruption_name` was needed only cause `disrupt_nodetool_seed_decommission` was calling this nemesis function

```suggestion
    def disrupt_nodetool_decommission(self, add_node=True):
        return self._nodetool_decommission(add_node=add_node)
```

in general out disrupt function shouldn't have parameters, and shouldn't be calling each other directly

Comment (fgelcer) [reply sdcm/nemesis.py]: fixed

**Review by fruch — APPROVED** [review]
LGTM

File: sdcm/nemesis.py [hunk 3]
```diff
@@ -1186,8 +1186,8 @@ def _terminate_cluster_node(self, node):
         self.cluster.terminate_node(node)
         self.monitoring_set.reconfigure_scylla_monitoring()
 
-    def disrupt_nodetool_decommission(self, add_node=True, disruption_name=None):
```
Comment (yarongilor) [inline sdcm/nemesis.py:1189]: cannot remove `disruption_name` parameter

Comment (fgelcer) [reply sdcm/nemesis.py:1189]: why? it was used only from the seed node decommission

**Comment by yarongilor** [comment]
> g job using this branch, run only this nemesis in the 1tb job (wait some 5 iterations of it), and if it solved the issue, we can merge this one.

testing it in: https://jenkins.scylladb.com/job/enterprise-2022.2/job/longevity/job/Reproducers/job/longevity-1tb-7days-test-seed-decommission/3/

**Comment by fgelcer** [comment]
> > g job using this branch, run only this nemesis in the 1tb job (wait some 5 iterations of it), and if it solved the issue, we can merge this one.
> 
> testing it in: https://jenkins.scylladb.com/job/enterprise-2022.2/job/longevity/job/Reproducers/job/longevity-1tb-7days-test-seed-decommission/3/

@yarongilor , it has an issue related to soft limit you introduced in another PR:
``` < t:2023-01-17 20:13:30,190 f:base.py         l:320  c:sdcm.sct_events.base p:WARNING > [SCT internal warning] (SoftTimeoutEvent Severity.ERROR) period_type=one-time event_id=64415293-1c1c-45ac-8727-b8a5e8bbddc7 during_nemesis=NodetoolSeedDecommission, source=SoftTimeout has not been published or dumped, maybe you missed .publish()
```

**Comment by yarongilor** [comment]
The nemesis passed ok 3 times, and started ok for the 4th time. aborting the test now.
image

**Pull request merged**